### PR TITLE
issue 533 : allow to use 0 padding in axis labels

### PR DESCRIPTION
**Pull request by alixdamman**

### CODE REVIEW

File: larray/util/misc.py [hunk 1]
```diff
@@ -515,8 +516,28 @@ def error_handler(error, flag):
...
+    >>> _isintstring('12')
+    True
+    >>> _isintstring('-12')
+    True
+    >>> _isintstring('a1')
+    False
+    >>> _isintstring('01')
+    False
+    """
+    return (s.isdigit() or (len(s) > 1 and s[0] == '-' and s[1:].isdigit())) and \
```
Comment (gdementen) [inline larray/util/misc.py]: This is not very readable IMO and will not work for negative zero padded ints. What do you think of the following?
```python
def isposint(s):
    # avoid zero padded strings
    return s.isdigit() and not (len(s) > 1 and s[0] == '0')
return isposint(s) or (len(s) > 1 and s[0] == '-' and isposint(s[1:]))
```

File: doc/source/changes/version_0_28.rst.inc [hunk 2]
```diff
@@ -320,6 +320,12 @@ Miscellaneous improvements
 
 * Renamed sheet_name of LArray.to_excel to sheet since it can also be an index (closes :issue:580).
 
+* allowed to create an axis with int-like labels starting with one or several 0 (closes :issue:`533`):
```
Comment (gdementen) [inline doc/source/changes/version_0_28.rst.inc]: int-like string labels?

File: larray/util/misc.py [hunk 3]
```diff
@@ -515,8 +516,28 @@ def error_handler(error, flag):
     return error_handler
 
 
+_zero_padding_pattern = re.compile('0+[1-9]+')
+
+
 def _isintstring(s):
-    return s.isdigit() or (len(s) > 1 and s[0] == '-' and s[1:].isdigit())
+    """
```
Comment (gdementen) [inline larray/util/misc.py]: explicitly state that zero padded strings are not considered to be int strings?

File: larray/util/misc.py (hunk 1 above)
Comment (alixdamman) [reply larray/util/misc.py]: Much better indeed. Thanks.

File: larray/core/group.py [hunk 4]
```diff
@@ -394,6 +402,10 @@ def _to_ticks(s, parse_single_int=False):
     ['U']
     >>> list(_to_ticks('..3'))
     [0, 1, 2, 3]
+    >>> _to_ticks('01..12')
+    ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12']
+    >>> _to_ticks('01,02,03,10,11,12')
+    ['01', '02', '03', 10, 11, 12]
```
[the remark this answers is not in the document]
Comment (alixdamman) [reply larray/core/group.py]: We may make `_to_ticks` return a Numpy array if there is no "Group" object in the list of ticks?

[the remark this answers is not in the document]
Comment (gdementen) [reply larray/core/group.py]: IIRC the "if" condition is not necessary as in recent versions of larray the **result** of _to_ticks **never** contains Group objects.

[the remark this answers is not in the document]
Comment (alixdamman) [reply larray/core/group.py]: If I understand correctly, the following piece of code:
```python
            ticks = _to_ticks(labels, parse_single_int=True)
            if _contain_group_ticks(ticks):
                # avoid getting a 2d array if all LGroup have the same length
                labels = np.empty(len(ticks), dtype=object)
                # this does not work if some values have a length (with a valid __len__) and others not
                # labels[:] = ticks
                for i, tick in enumerate(ticks):
                    labels[i] = tick
            else:
                labels = np.asarray(ticks)
```
in `Axis.labels` setter is now useless?

File: larray/core/group.py [hunk 5]
```diff
@@ -382,49 +388,60 @@ def _to_ticks(s, parse_single_int=False):
 
     Notes
     -----
-    This function is only used in Axis.__init__ and union().
+    1) This function is only used in Axis.__init__ and union().
+    2) In Axis.labels setter, the line labels = np.asarray(ticks) returns an Numpy array
```
Comment (gdementen) [inline larray/core/group.py]: outdated comment

File: doc/source/changes/version_0_28.rst.inc [hunk 6]
```diff
@@ -367,3 +373,13 @@ Fixes
     b0    0   3
     b1    1   4
     b2    2   5
+
+* fixed aggregation on arrays containing zero padded string labels (closes :issue:`522`):
+
+    >>> arr = ndtest(Axis('zero_padding=01,02,03,10,11,12'))
```
Comment (gdementen) [inline doc/source/changes/version_0_28.rst.inc]: My comment still stands. Why not, the simpler:

```python
ndtest('zero_padding=01,02,03,10,11,12')
```

File: larray/core/axis.py [hunk 7]
```diff
@@ -197,7 +197,7 @@ def labels(self, labels):
                 for i, tick in enumerate(ticks):
```
Comment (gdementen) [inline larray/core/axis.py]: Have you tried removing the _contain_group_ticks(ticks) code?

File: larray/util/misc.py [hunk 8]
```diff
@@ -7,6 +7,7 @@
 import math
 import itertools
 import sys
+import re
```
Comment (gdementen) [inline larray/util/misc.py]: unused import?

File: larray/core/axis.py (hunk 7 above)
Comment (alixdamman) [reply larray/core/axis.py]: All tests are green but I am unsure this case (i.e. ticks contain one or several groups) is actually tested somewhere.

Comment (gdementen) [reply larray/core/axis.py]: It was, but the tests have either been removed or commented since this is no longer supported AFAIK

**Pull request merged**